Utilities/AppleEfiSignTool/AppleEfiSignTool.c

```diff
@@ -44,34 +45,43 @@ PrintHelp (
     "AppleEfiSignTool v%a – Tool for verifying Apple EFI binaries\n",
     APPLE_EFI_SIGN_TOOL_VERSION
     ));
-  DEBUG ((DEBUG_ERROR, "It supports PE and Fat binaries.\n"));
+  DEBUG ((DEBUG_ERROR, "Supports PE and Fat binaries.\n"));
+  DEBUG ((DEBUG_ERROR, "Displays signing info. Displays APFS info if image is APFS driver.\n"));
+  DEBUG ((DEBUG_ERROR, "Also shows PE COFF image fixups when -f is specified.\n"));
 
-  DEBUG ((DEBUG_ERROR, "Usage: ./AppleEfiSignTool <path/to/image>\n"));
-  DEBUG ((DEBUG_ERROR, "Example: ./AppleEfiSignTool path/to/apfs.efi\n"));
+  DEBUG ((DEBUG_ERROR, "\nUsage: ./AppleEfiSignTool [-f] <path/to/image>\n"));
+  DEBUG ((DEBUG_ERROR, " -f force FixupAppleEfiImages quirk\n"));
+
+  DEBUG ((DEBUG_ERROR, "\nExample: ./AppleEfiSignTool path/to/apfs.efi\n"));
 }
 
 /**
-  Verify PE image signature.
+  Verify PE image signature. Report APFS info if present.
 
   @param[in,out]  Image      A pointer to image file buffer.
   @param[in]      ImageSize  Size of Image.
-  @param[out]     IsFat      Will be set to TRUE if Image is a FAT binary.
+  @param[out]     IsFat      Will be set to TRUE if Image is a FAT binary, or left unchanged otherwise.
   @param[in]      Arch       Image architechure.
 
   @return  0 if Image is successfully verified, otherwise EXIT_FAILURE.
 **/
 STATIC
 INT32
-VerifySignature (
+VerifySignatureAndApfs (
   IN  OUT  UINT8          *Image,
   IN       UINT32         ImageSize,
-  OUT  BOOLEAN            *IsFat,
+  IN OUT   BOOLEAN        *IsFat,
+  IN       BOOLEAN        ForceFixup,
   IN       PE_IMAGE_ARCH  Arch
   )
 {
-  EFI_STATUS   Status;
-  UINT32       OrgImageSize;
-  CONST CHAR8  *Slice;
+  EFI_STATUS                    Status;
+  EFI_STATUS                    ApfsStatus;
+  UINT32                        OrgImageSize;
+  CONST CHAR8                   *Slice;
+  PE_COFF_LOADER_IMAGE_CONTEXT  Context;
+  RETURN_STATUS                 ContextStatus;
```

Comment:
Possibly better to use EFI_STATUS type in EFI context? (I understand EFI_STATUS is superset of RETURN_STATUS.)

Reply:
But why? For the most part, there is nothing EFI about image processing.

Reply:
@mhaeuser - Well, because this very RETURN_STATUS is processed as EFI_STATUS in existing OC code at https://github.com/acidanthera/OpenCorePkg/blob/master/Library/OcPeCoffExtLib/OcPeCoffExtLib.c#L394 and https://github.com/acidanthera/OpenCorePkg/blob/master/Library/OcPeCoffExtLib/OcPeCoffExtLib.c#L483, and although processing RETURN_STATUS as RETURN_STATUS seemed more sane to me, I wanted to check I wasn't missing anything!